File: algorithm/sort/select/select.js
//선택정렬
function selectionSort(arr)
{
    var len = arr.length;
    var minidx = 0;

    for(var i = 0; i < len - 1; i++){
        for(var j = 1 + i, minidx = i; j < len; j++){
            if(arr[minidx] > arr[j]){
               minidx = j;
            }
        }
        if(minidx != i){
            var temp = arr[i];
            arr[i] = arr[minidx];
            arr[minidx] = temp;
        }
    }
    return arr;
}

var unsortedArray = [64, 34, 25, 12, 22, 11, 90];
console.log("정렬 전 : ", unsortedArray);
//선택 정렬
var sortedArray = selectionSort(unsortedArray);

console.log("정렬 후 : ", sortedArray);

File: algorithm/sort/select/select.c
#include <stdio.h>
#include <string.h>

void swap(char* a, char* b) {
    int temp = *a;
    *a = *b;
    *b = temp;
}

void QuickSort(char* ar, int size) {
// 최소값의 인덱스를 저장한 인덱스 선언
    int minidx = 0;

    for(int i = 0; i < size - 1; i++){
        minidx = i;
        for(int j = 1 + i; j < size; j++) {            
            if (ar[minidx] > ar[j]) {
                minidx = j;
            }
        }
        if(minidx != i) {
            swap(&ar[minidx], &ar[i]);
        }
    }
}

int main () 
{
    char str[] = "winapi";

    int len = strlen(str); // 6

    printf("정렬 전 문자열 : %s\n", str);


    // SelectionSort(str, len);
    // // 정렬 후 aiinpw
    // printf("정렬 후 문자열 : %s", str);

    return 0;
}